src/Compiler/Checking/CheckDeclarations.fs

```diff
@@ -385,6 +385,54 @@ let CheckNamespaceModuleOrTypeName (g: TcGlobals) (id: Ident) =
     // type names '[]' etc. are used in fslib
     if not g.compilingFSharpCore && id.idText.IndexOfAny IllegalCharactersInTypeAndNamespaceNames <> -1 then 
         errorR(Error(FSComp.SR.tcInvalidNamespaceModuleTypeUnionName(), id.idRange))
+        
+let CheckDuplicatesArgNames (synVal: SynValSig) m =
+    let argNames =
+        synVal.SynInfo.ArgNames
+        |> List.countBy id
+        |> List.filter (fun (_, count) -> count > 1)
+        |> List.map fst
+    for name in argNames do
+        errorR(Error((FSComp.SR.chkDuplicatedMethodParameter(name), m)))
+        
+let CheckDuplicateMemberNames (typeInfo: SynComponentInfo) (synMembersSig: SynMemberSig list) m=
+    let (SynComponentInfo(longId = longId)) = typeInfo
+    let synMembers =
+        synMembersSig
+        |> List.choose(fun sms ->
+            match sms with
+            | SynMemberSig.Member(memberSig = synValSig) ->
+                 match synValSig with
+                 | SynValSig(ident = SynIdent(ident, _)) ->
+                     Some ident.idText
+            | _ -> None)
+        
+    let memberNames =
+        synMembers
+        |> List.countBy id
+        |> List.filter (fun (_, count) -> count > 1)
+        |> List.map fst
```

Comment:
synMembers |> List.duplicates

(List.duplicates is already present in the compiler codebase)

Reply:
Updated :)